src/services/premise-validator.service.js

```diff
@@ -0,0 +1,44 @@
+const request = require('request');
+const winston = require('winston');
+
+const apiKey = process.env.PREMISE_VALIDATOR_API_KEY
+
+const baseUrl = 'https://api.ordnancesurvey.co.uk/places/v1/addresses/postcode?';
+
+const randomlyFail = () => Math.random() > 0.9;
+
+const validate = data => {
```

Comment:
Yeah like I said, pretty sure this works, but in general we should avoid editing data which has been passed in as a parameter - can create confusing object mutation situations: https://eslint.org/docs/rules/no-param-reassign

The way around this would be to do... `const output = [];
data.forEach(org => {
  // get validation data
  validatedOrg = Object.assign({}, org); 
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
  validatedOrg.addressValidated = whatever you got from results
  output.push(validatedOrg);
})
resolve(output)
`

So in general the pattern should be:

- Create new output array / object
- Iterate through input param, copy contents + new data to output
- Return output

Makes your function cleaner as it doesn't modify the input, so you can use it wherever without fear of your original object being modified.